skimage/filters/_gaussian.py

```diff
@@ -116,3 +115,34 @@ def gaussian(image, sigma=1, output=None, mode='nearest', cval=0,
     image = convert_to_float(image, preserve_range)
     return ndi.gaussian_filter(image, sigma, mode=mode, cval=cval,
                                truncate=truncate)
+
+
+def _guess_spatial_dimensions(image):
```

Comment:
I think we should take a chance and deprecate this completely from `gaussian`.

Reply:
Agreed. No more guessing! My apologies for writing that function, as I was young and naive... =)

Reply:
What, that's like me apologizing for writing the first version of scikit-image ;)

Reply:
Having said that, I don't think this should hold up this PR, which is already an improvement!

Reply:
@stefanv it would have certainly saved me lots of headaches if you hadn't. =P

Reply:
A conversation I have with myself quite frequently :joy:

Reply:
:laughing: